### PR TITLE
fix: improve studio error state ui

### LINKED ISSUE #14686: Studio prompts user to enter url even if an error has already occured
### Current behavior
If a test has failed and the user has not yet visit a url, they will still be prompted to enter a url.

Note: does not happen for failed `cy.visit(...)` commands as that was the only situation in which I ran into this during initial development and fixed.

### Desired behavior
Studio should not prompt the user to enter a url if their test has already failed

### Test code to reproduce
Attempting to extend the following suite produces this error:
```js
describe('studio', () => {
  beforeEach(() => {
    throw new Error('My Error')
  })

  it('types', () => {

  })
})
```

Screen Shot 2021-01-21 at 10 51 58 PM

### LINKED ISSUE #14691: Cancel button should appear under error for failed tests in Studio
### What would you like?
The cancel button (that usually appears in the same line as the "Save Commands" button) should appear when a test has finished running in Studio mode even if the test does not pass.

When a test has passed and is actively being edited:

Screen Shot 2021-01-21 at 11 22 00 PM

### Why is this needed?
The controls on the right side are less noticeable and may be missed in this situation. When a test passes, this button is available in the bottom left so it only makes sense that it would if the test fails since the user is most likely looking for the quickest way to exit.

This is what that state currently looks like, with the only exit being the 'X' in the top right:

Screen Shot 2021-01-21 at 11 18 16 PM

### PR DESCRIPTION
**Pull request by panzarino**
- Closes #14691
- Closes #14686

### User facing changelog
Improves the user interface when using studio and a command fails

### Additional details
Improves studio error state in the following ways:
- Adds cancel button to left side of reporter on fail
- Adds additional error message explaining that studio can't add commands to failed tests
- Prevents visit url prompt from appearing on failed tests
- Changes iframe border from blue -> red
- Icon in studio "navbar" no longer indicates recording on fail
- Updates available commands modal
- Updates fail state when user enters a url and visit fails

### How has the user experience changed?

Before

ezgif-7-c2b5734f14c9

After

ezgif-7-52975cb39052

Before

Screen Shot 2021-01-26 at 11 23 08 AM

After

Screen Shot 2021-01-27 at 10 36 48 AM

When a user enters a url and that visit fails (partially introduced in #14738):

Before

Screen Shot 2021-01-26 at 10 53 00 PM

After

Screen Shot 2021-01-26 at 11 09 49 PM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by jennifer-shehane — DISMISSED** [review]
I don't think this snapshot is capturing what you expected - it doesn't show the studio error. https://percy.io/cypress-io/cypress/builds-next/8535127/changed/485040355

Looks fine otherwise. Manually tested. The error state looks good. Is tested.

**Review by chrisbreiding — DISMISSED** [review]

**Comment by panzarino** [comment]
@jennifer-shehane Good catch on percy snapshot - fixed now

**Review by fornever847 — DISMISSED** [review]

**Comment by panzarino** [comment]
@jennifer-shehane also fixed modal docs issue you brought up here since I didn't want to make a separate PR along with an additional fix since the first review - more details in OP

File: packages/runner/src/studio/studio-modals.jsx [hunk 1]
```diff
@@ -26,11 +27,11 @@ export class StudioInstructionsModal extends Component {
...
-                <li><pre>cy.check()</pre></li>
-                <li><pre>cy.click()</pre></li>
-                <li><pre>cy.select()</pre></li>
-                <li><pre>cy.type()</pre></li>
-                <li><pre>cy.uncheck()</pre></li>
+                <li><pre>cy.get(...).check()</pre></li>
+                <li><pre>cy.get(...).click()</pre></li>
+                <li><pre>cy.get(...).select()</pre></li>
+                <li><pre>cy.get(...).type()</pre></li>
+                <li><pre>cy.get(...).uncheck()</pre></li>
```
Comment (jennifer-shehane) [inline packages/runner/src/studio/studio-modals.jsx:34]: No prefix for child commands. 

```suggestion
                <li><pre>.check()</pre></li>
                <li><pre>.click()</pre></li>
                <li><pre>.select()</pre></li>
                <li><pre>.type()</pre></li>
                <li><pre>.uncheck()</pre></li>
```

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**